File: Photomania/AppDelegate+MOC.m
//
//  AppDelegate+MOC.m
//  Photomania
//
//  Created by admin on 26/10/15.
//  Copyright (c) 2015 A. All rights reserved.
//

#import "AppDelegate+MOC.h"
#import <CoreData/CoreData.h>

@implementation AppDelegate (MOC)
-(void)saveContext: (NSManagedObjectContext *)managedObjectContext{
    NSError *error = nil;
    if (managedObjectContext != nil){
        if ([managedObjectContext hasChanges] && ![managedObjectContext save: &error]){
            abort();
        }
    }
}

-(NSPersistentStoreCoordinator *)createPersistentStoreCoordinator{
    NSPersistentStoreCoordinator * coordinator= nil;
    NSManagedObjectModel * model = nil;
    NSError *error = nil;
    model = [self createManagedObjectModel];
    if (model != nil) {
        coordinator =  [[NSPersistentStoreCoordinator alloc]initWithManagedObjectModel:model];
    }
    NSURL* storeURL =  [[[[NSFileManager defaultManager] URLsForDirectory:NSDocumentDirectory inDomains:NSUserDomainMask] lastObject] URLByAppendingPathComponent:@"MOC.sqlite"];
    
    if (coordinator != nil){
        if (![coordinator addPersistentStoreWithType:NSSQLiteStoreType configuration:nil URL:storeURL options:nil error:&error]){
            NSLog(@"Unresolved error %@, %@", error, [error userInfo]);
            abort();
        }
    }
    return coordinator;
}
-(NSManagedObjectContext *)createMainQueueManagedObjectContext{
    NSManagedObjectContext *managedObjectContext = nil;
    NSPersistentStoreCoordinator *coordinator = [self createPersistentStoreCoordinator];
    if (coordinator != nil){
     managedObjectContext = [[NSManagedObjectContext alloc]initWithConcurrencyType:NSMainQueueConcurrencyType];
    [managedObjectContext setPersistentStoreCoordinator:coordinator];
    }
    
    return managedObjectContext;
}

-(NSManagedObjectModel *)createManagedObjectModel{
    NSManagedObjectModel *model = nil;
    NSURL * modelURL = [[NSBundle mainBundle] URLForResource:@"Model" withExtension:@"momd"];
    model = [[NSManagedObjectModel alloc]initWithContentsOfURL:modelURL];
    
    return model;
    
}
@end;

File: Smashtag/Smashtag/AppDelegate+MOC.swift
//
//  AppDelegate+MOC.swift
//  Smashtag
//
//  Created by admin on 05/11/15.
//  Copyright (c) 2015 A. All rights reserved.
//
import Foundation
import CoreData
extension AppDelegate {
    func createMainQueueManagedObjectContext() -> NSManagedObjectContext?{
        let modelURL = NSBundle.mainBundle().URLForResource("Model", withExtension:"momd")
        // The managed object model for the application. It is a fatal error for the application not to be able to find and load its model.
        let mom = NSManagedObjectModel(contentsOfURL: modelURL!)
        let psc = NSPersistentStoreCoordinator(managedObjectModel: mom!)
        self.managedObjectContext = NSManagedObjectContext(concurrencyType: .MainQueueConcurrencyType)
        self.managedObjectContext?.persistentStoreCoordinator = psc
        dispatch_async(dispatch_get_global_queue(DISPATCH_QUEUE_PRIORITY_BACKGROUND, 0)) {
            let urls = NSFileManager.defaultManager().URLsForDirectory(.DocumentDirectory, inDomains: .UserDomainMask)
            let docURL = urls[urls.endIndex-1]
            /* The directory the application uses to store the Core Data store file.
            This code uses a file named "DataModel.sqlite" in the application's documents directory.
            */
            let storeURL = docURL.URLByAppendingPathComponent("DataModel.sqlite")
            
            psc.addPersistentStoreWithType(NSSQLiteStoreType, configuration: nil, URL: storeURL, options: nil, error : NSErrorPointer())
        }
        return managedObjectContext
    }
}